### ISSUE TITLE
Predictions from provided tflite files of same Mobilenet model do not match

### ISSUE THREAD

**Issue by abhi-rf** [tensorflow/tensorflow#20627]
Hi Guys. I downloaded the Mobilenet_v1_1.0_224 and Mobilenet_v1_1.0_224_quant from the links provided in the documentation. I then use the tflite models provided in both of these for prediction using the tflite interpeter [Link](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/toco/g3doc/python_api.md#using-the-interpreter-from-a-model-file-) I am able to make the predictions. However, the tflite predictions from the 'quant' version are very random. The float tflite model makes the correct prediction while the quantised tfite model makes very weird predictions. I even comapred the answers to normal frozen models provided and the quantized tflite model does not match the predictions at all while the float tflite makes all the predictions same as the frozen .pb model. 

I then tried the same thing with different mobilenet versions but the same observations.
Kindly help.
------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: 1.10 dev
- **Python version**: 2.7
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: NA
- **GPU model and memory**: NA
- **Exact command to reproduce**:

### Describe the problem
I then use the tflite models provided in both of these for prediction using the tflite interpeter [Link](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/toco/g3doc/python_api.md#using-the-interpreter-from-a-model-file-) I am able to make the predictions. However, the tflite predictions from the 'quant' version are very random. The float tflite model makes the correct prediction while the quantised tfite model makes very weird predictions. I even comapred the answers to normal frozen models provided and the quantized tflite model does not match the predictions at all while the float tflite makes all the predictions same as the frozen .pb model. 

I then tried the same thing with different mobilenet versions but the same observations.

I am also making sure that I use the exact same images for both so that is not an issue just in case.
Kindly help.

### Source code / logs
img = np.array(PIL.Image.open(image_path).resize((224, 224))).astype(np.uint8) / 128 - 1

	img = img.reshape(1,224,224,3)
	#print (img.shape)
	
	start = time.time()

	interpreter.set_tensor(input_details[0]['index'], img)

	interpreter.invoke()
	end = time.time()

	output_data = interpreter.get_tensor(output_details[0]['index'])
	#output_data_1 = interpreter.get_tensor(output_details[1]['index'])

	#print(output_data.argmax(),output_data.max())
	label_map = imagenet.create_readable_names_for_imagenet_labels()  
	print("Top 1 Prediction: ", output_data.argmax(),label_map[output_data.argmax()], output_data.max(), k+1)

**Comment by aselle** [comment]
@suharshs, could you please handle this issue.

**Comment by aselle** [comment]
```
img = np.array(PIL.Image.open(image_path).resize((224, 224))).astype(np.uint8) / 128 - 1
```

this looks wrong as it will make the range -1 to 1 . you want to feed 0,255 which is probably what it was
so try feeding
```
img = np.array(PIL.Image.open(image_path).resize((224, 224))).astype(np.uint8)
```
but check by checking mins and maxes of the img array once done ...

**Comment by suharshs** [comment]
I am closing this issue due to inactivity, if aselle's instructions do not work, please reopen. Thanks!

**Comment by suharshs** [comment]
Sure, could you create an new issue describing what you are seeing and how to reproduce the latency numbers?

**Comment by abhi-rf** [comment]
Sure. Thanks a lot

**Comment by smohan10** [comment]
Hi, 
I downloaded the tar file of Mobilenet v1.0-224 from the list of hosted quantized TF .pb and .lite model. I am unable to get the same numbers when I run inference for .pb model and .tflite model. 

I am feeding image net data resized to (224,224,3). When I run inference and check the argmax of the output vector, the indices of .pb model and .tflite model are different. 

I did use image.astype(np.uint8). When I print the interpreter's input and output details, I do see a tuple of numbers in the quantization key of the tensor dictionary. 
For example: 
[{'name': 'input', 'shape': array([  1, 299, 299,   3], dtype=int32), 'dtype': <class 'numpy.uint8'>, 'quantization': (0.0078125, 128), 'index': 315}]
 [{'name': 'output', 'shape': array([   1, 1001], dtype=int32), 'dtype': <class 'numpy.uint8'>, 'quantization': (0.07597498595714569, 53), 'index': 316}]

How do I use this information to match the results of the .pb and .tflite model? Any advice would be really appreciated!
